### ISSUE TITLE
Update Security Groups FAQ

### ISSUE THREAD

**Issue by trentgillaspie** [rackerlabs/rackspace-how-to#1413]
https://support.rackspace.com/how-to/cloud-networks-faq/

Where links currently exist in the text in the FAQ right now, please include them accordingly in the text below, along with any formatting that may currently be in place within the FAQ online documentation. Please let me know if you have any questions, or concerns. Thank you! - Trent

-----------------
 
What are security groups?
Security groups are containers for a set of inbound and outbound traffic rules that are directly applied to a Neutron port (PublicNet, ServiceNet or Cloud Network).
 
What is the difference between security groups and a firewall?
Security groups act as a distributed firewall for your Cloud Server instances. After you launch an instance, you can assign one or more security groups to ports on that instance.
 
Where can I find documentation?
Cloud Networks Developer Guide
Cloud Networks API Reference
 
What are the benefits of using security groups?
Prior to the release of this feature, users had to manage traffic to and from their instances individually via iptables rules (as an example) on every instance, or perhaps use third-party tools such as CloudPassage. Managing firewall policies involves significant overhead. Security groups make it possible to use a self-service API to define a common set of rules and apply them to the servers without needing to manage iptables rules on each server. Security groups simplify security policy administration for customers across their deployments.
 
Is this feature available to all Cloud customers?
No. This feature is currently in limited availability in all data centers. Please contact Rackspace support or your Account Manager to request access.
 
What limitations does the feature currently have?
Cannot add security group as a child of another security group. You cannot edit a security group rule - you have to add a new one to replace the old one.
 
Are security groups supported via the neutron client?
Yes. Users can provision security groups via the neutron client.

Is this functionality integrated with and available from the Cloud Control Panel?
Yes, but only inbound PublicNet and ServiceNet security groups are currently available in the control panel. To access additional features, you can use either the neutron client or the API.
 
Are security groups supported for OnMetal users?
No. We currently support security groups only for virtual cloud servers.
 
Is a default security group applied to my instances?
No default security groups are applied. Users must create a security group themselves and apply it to ports on an instance.
 
Can I apply security groups to ports on an instance at boot time?
No. Security groups can be applied only after the instance is active. We are actively working to include this feature in an upcoming release.
 
What happens when a security group rule is added to the security group?
Traffic that matches the new security group rule is allowed to go through.
 
Can traffic be blocked or denied based on a security group rule?
No. Traffic that matches a rule is permitted, and any traffic that is not part of the ruleset for that security group is denied or blocked. Because of OpenStack API design requirements, you cannot specify that traffic matching a rule should be denied. The security groups API is a whitelist. Thus, traffic that doesn’t match any of the rules in the whitelist is automatically blacklisted.
 
Is there any traffic that is permitted or allowed by default by security groups?
Very basic network operations are allowed by default such as DNS responses from Rackspace Provider DNS servers (UDP source port 53) are allowed by default even if a security group does not explicitly allow them. In addition, TCP flags ACK and RST are permitted by default.
 
What kinds of traffic can be matched by the security group rules?
The following types of traffic can be matched (for both IPv4 and IPv6 addresses):
TCP traffic
UDP traffic
ICMP traffic
Traffic from a Source IP address
Traffic from a CIDR
 
Can I have a security group with no rules?
Yes. Such a security group will deny or block all traffic.
 
Are security groups applied to instances?
No. Security groups are applied to a Neutron port on a network that is attached to an instance and not to an instance itself. In the near future, when updating a security group, changes will be applied automatically to any port associated with that security group.
 
What are the available quotas for security groups and rules?
You can apply up to 5 security groups per port.
You can have up to 20 security group rules per security group
You can have up to 100 security group rules (aggregate) per user during the limited availability release.
